packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-oidc.tsx

```diff
@@ -16,6 +17,8 @@ import { errorMessageByFieldName } from '../../../utils/validation';
 import { getConnectionStringUsername } from '../../../utils/connection-string-helpers';
 import type { OIDCOptions } from '../../../utils/oidc-handler';
 import { useConnectionFormPreference } from '../../../hooks/use-connect-form-preferences';
+import { usePreference } from 'compass-preferences-model/provider';
```

Comment:
So, we did a poor job documenting this well, but connection-form, being used by vscode too, uses this `use-connect-form-preferences` as a way to have a access to preferences inside the form so that we don't need to pull anything compass specific into runtime and depend on any compass logic for default preferences at all. There are a few places where preferences got used directly, and we already have ticket to clean those up, so let's add this as a new value to the connection form preferences to avoid adding more work for this planned cleanup

Reply:
So – I've stuck to this pattern because I've been using it for a feature flag only, similar to what we do for `enableMultipleConnectionSystem` (where we also call `usePreference()` in connection-form). The feature flag gets nuked in the next PR anyway, so I'd be tempted to just leave it like this? (Will address the other comments below tho)

Reply:
Hmmm, I get this reasoning, at the same time the multiple connection feature flag directly used from the preferences was one of the things that broke connection form for vscode based [on Alenas update attempt](https://jira.mongodb.org/browse/VSCODE-549?focusedCommentId=6571050&focusedId=6571050&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-6571050) investigation, so this case feels similar and it might be a bit inconvenient if periodically vscode couldn't update while feature flags are used in the form. The place where we map compass preferences to connection form ones is in [this file](https://github.com/mongodb-js/compass/blob/main/packages/compass-connections/src/hooks/use-connection-form-preferences.tsx), so I wonder that maybe the logic you have here where it derives the value for enableProxySettings from the form preference value and a feature flag can be combined there? Would that make sense to you?

Reply:
Any chance you linked the wrong file here? Anyway, I moved this out of connection-form in 03fe506fae0c328d76e4220a01869f5954c62292, I hope that looks like what you had in mind (and, again, this is very temporary 🙂)

Reply:
I don't think so, unless github is glitching out somehow, I'm linking to the same file where you moved the logic to though so it definitely looks good to me 😄 Sorry for being a nuisance with this, I'm planning to extend the eslint rules in this package shortly to cover more packages that we should avoid using directly here, so just trying to find some relatively easy ways for this PR to avoid these imports in advance